### ISSUE TITLE
Feature request: Table Render for Vue

### ISSUE THREAD

**Issue by retronew** [inokawa/virtua#456]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I need to display tabular data in a Vue component and there is no built-in support for a table mode. Instead, the component always outputs a `<div>` tag, which is not suitable for organizing and presenting tabular data effectively.

**Describe the solution you'd like**
I would like the component to have native support for a table render in Vue.

**Additional context**
Supporting dynamic HTML tags would enhance the component's versatility and compatibility, simplifying integration into Vue applications and improving the development experience.

**Comment by inokawa** [comment]
#430 is an issue to implement a prop to override the `div` of items (PR welcome).

However, although React virtualizer already have it, it wasn't good for table in some situations (#312).
So we are planning to create table virtualizer component for React. I'll also implement it for Vue!

**Comment by onx2** [comment]
You can display tabular data with divs pretty effectively with `display: grid;` You could try that approach while this feature isn't in 😄 

For example:

```html
<div class="grid-table">
    <div class="grid-header">
      <div class="grid-cell">Header 1</div>
      <div class="grid-cell">Header 2</div>
      <div class="grid-cell">Header 3</div>
    </div>
    <div class="grid-row">
      <div class="grid-cell">Row 1, Cell 1</div>
      <div class="grid-cell">Row 1, Cell 2</div>
      <div class="grid-cell">Row 1, Cell 3</div>
    </div>
    <div class="grid-row">
      <div class="grid-cell">Row 2, Cell 1</div>
      <div class="grid-cell">Row 2, Cell 2</div>
      <div class="grid-cell">Row 2, Cell 3</div>
    </div>
    <div class="grid-row">
      <div class="grid-cell">Row 3, Cell 1</div>
      <div class="grid-cell">Row 3, Cell 2</div>
      <div class="grid-cell">Row 3, Cell 3</div>
    </div>
  </div>
```
```css
.grid-table {
  display: grid;
  grid-template-columns: repeat(3, 1fr); /* 3 columns with equal width */
  gap: 1px; /* Gap between cells */
  background-color: #ccc; /* Background color for the gaps */
}

.grid-header {
  display: contents; /* Use contents to make sure the header cells align with the grid */
}

.grid-row {
  display: contents; /* Use contents to make sure the row cells align with the grid */
}

.grid-cell {
  padding: 10px;
  background-color: white;
  border: 1px solid #ccc;
  text-align: center;
}
```
<img width="877" alt="image" src="https://github.com/inokawa/virtua/assets/23039725/4c8f40e2-76ac-4524-bb7c-6fd639bb8c13">


This is a contrived example, but I just wanted to show how it is possible to use `div` instead of `table` - hope it helps!

**Comment by onx2** [comment]
I just tested making a table in solid and I see what you mean with it being difficult. With the current approach, the component wraps everything in div and isn't able to be styled easily. Would love to see some more flexibility as well 😄

**Comment by inokawa** [comment]
From `0.33.2`, you can implement table virtualization with virtua, similar to React or Solid adapter. So close this issue but any feedback is welcome.
https://github.com/inokawa/virtua/blob/main/stories/vue/TableElement.vue

And in the future, a component like `VTable` may be implemented in #312 for easier integration.

**Comment by inokawa** [comment]
I've released `VGrid` component in `0.52.0`. I recommend using it since table support of `Virtualizer` is not perfect (https://github.com/inokawa/virtua/issues/312#issuecomment-1881101480).
